docs/github.md

```diff
@@ -91,7 +92,23 @@ $ jj commit -m 'address pr comments'
 $ # Update the branch to point to the new commit.
 $ jj branch set your-feature -r @-
 $ # Push it to your remote
-$ jj git push.
+$ jj git push
+```
+
+Notably, the above workflow creates a new commit for you. The same can be 
+achieved without creating a new commit. 
+
+```shell script
+$ # Create a new commit on top of the `your-feature` branch from above.
+$ jj new your-feature
+$ # Address the comments, by updating the code
+$ jj diff
+$ # Give the fix a description.
+$ jj describe -m 'address pr comments'
+$ # Update the branch to point to the current commit.
+$ jj branch set your-feature -r @
+$ # Push it to your remote
+$ jj git push
```

Comment:
I'd rather leave this workflow out because it's riskier (if the user forgets that they had not created a new working copy for their next feature, they might accidentally add that to the current commit). I think it's better to teach people the safer workflow and then they can learn other workflows and decide for themselves once they're more familiar with jj.

Also, this says "The same can be achieved without creating a new commit." and then the first step is "Create a new commit..." :)

Reply:
> I'd rather leave this workflow out because it's riskier (if the user forgets that they had not created a new working copy for their next feature, they might accidentally add that to the current commit).

I agree, but this time it was an explicit request that it exists :-/. 

> I think it's better to teach people the safer workflow and then they can learn other workflows and decide for themselves once they're more familiar with jj.

Many parts of #1989 are arguably about a workflow and as there is no definitive way of using `jj` yet, it is appropriate here. If there was one, I'd point the users to it and close the issue. 

> Also, this says "The same can be achieved without creating a new commit." and then the first step is "Create a new commit..." :)

Well we drop the new commit from `jj commit` but there rest should stay the same.

Reply:
I wouldn't recommend this workflow either: it becomes confusing dangerous if the PR is more than one commit long.

Instead, we could rewrite the following section to recommend a `jj split` + `jj move` (or `rebase` & `squash`) workflow. But it doesn't have to be in this PR necessarily.

Reply:
So it sounds like Ilya and I are against this section and you and Joy are for it. Anyone else has any opinion?

Reply:
Actually, I misunderstood what this was about completely. At least, I can't remember my thoughts when writing the previous comment in a way that makes sense. I think I was thinking about whether you squash the fixes into the old commit or create a new commit for the fixes.

It seems the only difference here is the *empty* commit that's created in the first workflow.

I'm still leaning against including this, though. In my mind, it's not different enough from the first version to bother to explain both. Perhaps we could just say something along these lines:

> (Workflow 1)
>
> Now, the working copy would be on a new empty commit where you can start working on new fixes. Alternatively, if you switch to another task and do `jj new <another commit>` before making any changes to the working copy or adding a description to the empty commit, it will be removed automatically. This avoids cluttering your repository with useless empty commits.

Reply:
I've added a markdown warning to the section and incorporated #2301 changes into a new section. I am partially  satisfied with the current solution. 

> **Warning**
> Like this.